python/python/lance/arrow.py

```diff
@@ -589,13 +587,15 @@ def _from_sequence(
             return PandasBFloat16Array(bfloat16_array(scalars))
 
         def __getitem__(self, item):
-            import numpy as np
-
             if isinstance(item, int):
                 return self.data[item].as_py()
             elif isinstance(item, slice):
                 return PandasBFloat16Array(self.data[item])
-            elif isinstance(item, np.ndarray) and item.dtype == bool:
+            elif (
+                _check_for_numpy(item)
```

Comment:
Why do you need to pass in an object when calling a `_check_for_xyz` function?

Reply:
The `_check_for_xxx` is checking both if the library exist and also if the object potentially is a type from this library (following the `_might_be` discussion above).